dev/Breadcrumb/BreadcrumbItem.cpp

```diff
@@ -0,0 +1,263 @@
+        newItemsSource.Append(item);
+    }
+
+    return newItemsSource;
+}
+
+void BreadcrumbItem::OpenFlyout()
+{
+    if (auto flyout = m_ellipsisFlyout.get())
+    {
+        if (SharedHelpers::IsFlyoutShowOptionsAvailable())
+        {
+            winrt::FlyoutShowOptions options{};
+            flyout.ShowAt(*this, options);
+        }
+        else
+        {
+            flyout.ShowAt(*this);
+        }
+    }
+}
+
+void BreadcrumbItem::CloseFlyout()
+{
+    if (auto flyout = m_ellipsisFlyout.get())
+    {
+        flyout.Hide();
+    }
+}
+
+void BreadcrumbItem::OnEllipsisItemClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (const auto& breadcrumb = m_parentBreadcrumb.get())
+    {
+        if (const auto& breadcrumbImpl = breadcrumb.try_as<Breadcrumb>())
+        {
+            const auto& hiddenElements = CloneEllipsisItemSource(breadcrumbImpl->HiddenElements());
+
+            if (const auto& flyoutRepeater = m_ellipsisItemsRepeater.get())
+            {
+                flyoutRepeater.ItemTemplate(m_ellipsisDataTemplate.get());
+                flyoutRepeater.ItemsSource(hiddenElements);
+            }
+
+            OpenFlyout();
+        }
+    }
+}
+
+void BreadcrumbItem::SetPropertiesForLastNode()
+{
+    m_isEllipsisNode = false;
+    m_isLastNode = true;
+
+    winrt::VisualStateManager::GoToState(*this, L"LastNode", false);
+}
+
+void BreadcrumbItem::ResetVisualProperties()
+{
+    m_isEllipsisNode = false;
+    m_isLastNode = false;
+
+    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        breadcrumbItemButton.Flyout(nullptr);
+    }
+    m_ellipsisFlyout.set(nullptr);
+    m_ellipsisItemsRepeater.set(nullptr);
+
+    winrt::VisualStateManager::GoToState(*this, L"Normal", false);
+}
+
+void BreadcrumbItem::InstantiateFlyout()
+{
+    // Only if the element has been created visually, instantiate the flyout
+    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        // Create ItemsRepeater and set the DataTemplate 
+        const auto& ellipsisItemsRepeater = winrt::ItemsRepeater();
+        ellipsisItemsRepeater.Name(L"PART_EllipsisItemsRepeater");
+        ellipsisItemsRepeater.HorizontalAlignment(winrt::HorizontalAlignment::Stretch);
+
+        if (const auto& dataTemplate = m_ellipsisDataTemplate.get())
+        {
+            ellipsisItemsRepeater.ItemTemplate(dataTemplate);
+        }
+
+        m_ellipsisRepeaterElementPreparedRevoker = ellipsisItemsRepeater.ElementPrepared(winrt::auto_revoke, { this, &BreadcrumbItem::OnFlyoutElementPreparedEvent });
+        
+        m_ellipsisItemsRepeater.set(ellipsisItemsRepeater);
+
+        // Create the Flyout and add the ItemsRepeater as content
+        const auto& ellipsisFlyout = winrt::Flyout();
+        ellipsisFlyout.Content(ellipsisItemsRepeater);
+        ellipsisFlyout.Placement(winrt::FlyoutPlacementMode::Bottom);
```

Comment:
Does this work when the Breadcrumb is at the bottom of the app?

Reply:
When the breadcrumb is at the bottom the flyout is show on top of the BreadcrumbItem

![image](https://user-images.githubusercontent.com/35784165/104515200-53dfb000-55a7-11eb-8028-95de03de5948.png)

Reply:
I hope you are also testing Right to Left support :)